README.md

```diff
@@ -1,30 +1,54 @@
+
+Cordis is cut up in the form of multiple packages/libraries:
+
+- [`@cordis/bitfield`](https://github.com/cordis-lib/cordis/tree/main/libs/bitfield) - A simple structure for working with bitfields using BigInts
+- [`@cordis/brokers`](https://github.com/cordis-lib/cordis/tree/main/libs/brokers) - Message brokers for routing patterns, pub/sub systems and RPC!
+- [`@cordis/common`](https://github.com/cordis-lib/cordis/tree/main/libs/common) - Common structures and utility shared across packages.
+- [`@cordis/gateway`](https://github.com/cordis-lib/cordis/tree/main/libs/gateway) - A very flexible way of connecting to Discord's gateway, with clustering support.
+  - [`@cordis/gateway-service`](https://github.com/cordis-lib/cordis/tree/main/services/gateway) ([Docker](https://hub.docker.com/r/cordislib/gateway)) - Service that hooks into RabbitMQ using `@cordis/gateway`
+- [`@cordis/queue`](https://github.com/cordis-lib/cordis/tree/main/libs/queue) - A simple and compact sequential queue for async operations
+- [`@cordis/redis-store`](https://github.com/cordis-lib/cordis/tree/main/libs/redis-store) - A Redis implementation of `@cordis/store`
+- [`@cordis/rest`](https://github.com/cordis-lib/cordis/tree/main/libs/rest) - Tooling for making HTTP requests to Discord, with rate limiting handling
+- [`@cordis/routers`](https://github.com/cordis-lib/cordis/tree/main/libs/routers) - Make API requests with ease using simple dynamic JavaScript property accessing
+- [`@cordis/snowflake`](https://github.com/cordis-lib/cordis/tree/main/libs/snowflake) - A simple snowflake structure for destructuring Discord IDs into relevant information
+- [`@cordis/store`](https://github.com/cordis-lib/cordis/tree/main/libs/store) - A simple map-like interface for holding key-value pairs - ships with an in-memory implementation
 
-It is cut up in multiple packages/libraries:
+# Contributing
+We make use of [`PNPM`](https://pnpm.js.org/) to manage our monorepo setup. It is expected that you have an up-to-date version of it. 
 
-- `@cordis/bitfield`, for working with bitfields using BigInts
-- `@cordis/brokers`, message brokers for routing patterns, pub/sub systems and RPC!
-- `@cordis/common`, cordis core utilities
-- `@cordis/gateway`, containing a very flexible way of connecting to Discord's gateway - with clustering support
-- `@cordis/queue`, simple, small sequential queue for async operations
-- `@cordis/redis-store`, Redis implementation of `@cordis/store`
-- `@cordis/rest`, tooling for making HTTP requests to Discord, with rate limiting handling
-- `@cordis/routers`, make API requests with ease using simple dynamic javascript property accessing
-- `@cordis/snowflake`, simple snowflake structure for destructuring Discord IDs into relevant information
-- `@cordis/store`, Simple map-like interface for holding key-value pairs - ships with an in-memory implementation
+Please ensure you run `pnpm run lint` and `pnpm run build` in the root before pushing your commits.
```

Comment:
Linting already runs on commit anyway assuming dependencies are installed. We could make build and test also run on that thought, thoughts?